todo-angular/src/app/todo/task/edit-task/edit-subtasks/edit-subtasks.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { SubTaskModel } from 'src/Models/TodoModel';
+
+@Component({
+  selector: 'edit-subtasks',
+  templateUrl: './edit-subtasks.component.html',
+  styleUrls: ['./edit-subtasks.component.scss'],
+})
+export class EditSubtasksComponent implements OnInit {
+  constructor() {}
+  hasSubTasks = false;
+  @Input() subTasks?: SubTaskModel[];
+  @Input() taskId?: number = 0;
+  @Output() subTasksUpdated: EventEmitter<SubTaskModel[]> = new EventEmitter<
+    SubTaskModel[]
+  >();
+
+  newSubTaskName = new FormControl('', {
+    nonNullable: true,
+    validators: [Validators.minLength(1)],
+  });
+
+  onSubTaskCompleted(subTask: SubTaskModel) {
+    console.log('subtask completed', subTask);
+
+    const subTaskIndex = this.subTasks?.findIndex(
+      (subTask) => subTask.subTaskId === subTask.subTaskId
+    );
+    if (subTaskIndex !== undefined) {
+      this.subTasks![subTaskIndex].subComplete = !subTask.subComplete;
```

Comment:
Like with Todos, maybe this should be emited, and only when the backend has succesfully updated the subtask, then we update it in the frontend :) If you feel this is too much to achieve before 13 o clcok, just leave it